File: src/main/java/com/muxiu1997/sharewhereiam/integration/Mods.java
package com.muxiu1997.sharewhereiam.integration;

import cpw.mods.fml.common.Loader;

public enum Mods {
    VISUAL_PROSPECTING("visualprospecting"),
    ;

    public final String modid;
    private boolean loaded;

    Mods(String modid) {
        this.modid = modid;
    }

    public boolean isLoaded() {
        return loaded;
    }

    public void check() {
        this.loaded = Loader.isModLoaded(modid);
    }
}

File: src/main/kotlin/com/muxiu1997/sharewhereiam/integration/Mods.kt
package com.muxiu1997.sharewhereiam.integration

import cpw.mods.fml.common.Loader


const val VISUAL_PROSPECTING_MODID = "visualprospecting"

enum class Mods(val modid: String) {
    VISUAL_PROSPECTING(VISUAL_PROSPECTING_MODID),
    ;

    var isLoaded = false
        private set

    fun check() {
        isLoaded = Loader.isModLoaded(modid)
    }
}
